src/backend/joanie/tests/core/api/order/test_create.py

```diff
@@ -697,6 +697,88 @@ def test_api_order_create_auto_assign_organization_with_least_orders(self):
         order = models.Order.objects.get(id=order_id)
         self.assertEqual(order.organization, organization_with_least_active_orders)
 
+    def test_api_order_create_get_organization_with_least_active_orders_prefer_author(
+        self,
+    ):
+        """
+        In case of order count equality, the method _get_organization_with_least_orders should
+        return first organization which is also an author of the course.
+        """
+        user = factories.UserFactory()
+        token = self.generate_token_from_user(user)
+
+        organization, expected_organization = (
+            factories.OrganizationFactory.create_batch(2)
+        )
+
+        relation = factories.CourseProductRelationFactory(
+            organizations=[organization, expected_organization]
+        )
+
+        relation.course.organizations.set([expected_organization])
+
+        # Create 3 orders for the first organization (1 draft, 1 pending, 1 canceled)
+        factories.OrderFactory(
+            organization=organization,
+            product=relation.product,
+            course=relation.course,
+            state=enums.ORDER_STATE_PENDING,
+        )
+        factories.OrderFactory(
+            organization=organization,
+            product=relation.product,
+            course=relation.course,
+            state=enums.ORDER_STATE_CANCELED,
+        )
+
+        # 2 ignored orders for the second organization (1 pending, 1 canceled)
+        factories.OrderFactory(
+            organization=expected_organization,
+            product=relation.product,
+            course=relation.course,
+            state=enums.ORDER_STATE_PENDING,
+        )
+        factories.OrderFactory(
+            organization=expected_organization,
+            product=relation.product,
+            course=relation.course,
+            state=enums.ORDER_STATE_CANCELED,
```

Comment:
This is no more relevant to add `PENDING` order. Instead we could create `Draft` and `Assigned` orders.